lib/house.rb

```diff
@@ -0,0 +1,33 @@
+class House
+  attr_reader :price,
+              :address,
+              :rooms
+
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    false
+  end
+
+  def rooms_from_category(category)
+    bedrooms = @rooms.find_all do |room|
+      if room.category == category
+        room
+      end
+    end
+  end
+
+  def area
+    @rooms.map do |room|
+      room.area
+    end
+  end
```

Comment:
Good start on this, you are on the right logical path.  Here are some ways you could write this method:

```ruby
def area
  @rooms.map do |room|
    room.area
  end.sum
end
```
```ruby
def area
  @rooms.sum do |room|
    room.area
  end
end
```
```ruby 
  def area
    house_area = 0
    @rooms.each do |room|
      house_area += room.area
    end
    house_area
  end
```